crates/wasm-compose/src/composer.rs

```diff
@@ -531,7 +554,8 @@ impl<'a> ComponentComposer<'a> {
     /// Returns the bytes of the composed component.
     pub fn compose(&self) -> Result<Vec<u8>> {
         let (root_instance, graph) =
-            CompositionGraphBuilder::new(self.component, self.config)?.build()?;
+            CompositionGraphBuilder::new(self.component, self.config, self.features)?
+                .build(self.features)?;
```

Comment:
To avoid threading `features` around throughout here, WDYT of using `WasmFeatures::all()` as the validation side of things? Personally I find it useful enough to defer specific features/flags/etc to the main validate step in an engine or similar.